src/identity/hcs/did/hcs-did.ts

```diff
@@ -0,0 +1,337 @@
+import {HederaDid} from "../../hedera-did";
+import {DidSyntax} from "../../did-syntax";
+import {FileId, PrivateKey, PublicKey, TopicId} from "@hashgraph/sdk";
+import {Hashing} from "../../../utils/hashing";
+import {DidDocumentBase} from "../../did-document-base";
+import {HcsDidRootKey} from "./hcs-did-root-key";
+
+/**
+ * Hedera Decentralized Identifier for Hedera DID Method specification based on HCS.
+ */
+export class HcsDid implements HederaDid {
+    public static DID_METHOD = DidSyntax.Method.HEDERA_HCS;
+    private static DID_PARAMETER_VALUE_PARTS = 2;
+
+    private didTopicId: TopicId;
+    private addressBookFileId: FileId;
+    private network: string;
+    private idString: string;
+    private did: string;
+    private didRootKey: PublicKey;
+    private privateDidRootKey: PrivateKey;
```

Comment:
Can these be final?

Reply:
typescript have not final modifier, I can make this fields readonly, but it is not same thing